Source/diablo.cpp

```diff
@@ -1808,113 +1812,115 @@ void __fastcall LoadGameLevel(BOOL firstflag, int lvldir)
+				DeltaLoadLevel();
+
 			IncProgress();
-		LABEL_55:
-			if (gbMaxPlayers == 1)
-				ResyncQuests();
-			else
-				ResyncMPQuests();
-			goto LABEL_72;
 		}
+		if(gbMaxPlayers == 1)
+			ResyncQuests();
+		else
+			ResyncMPQuests();
+	} else {
+		/// ASSERT: assert(! pSpeedCels);
+		pSpeedCels = DiabloAllocPtr(0x100000);
+		LoadSetMap();
+		IncProgress();
+		GetLevelMTypes();
+		InitMonsters();
+		InitMissileGFX();
+		InitDead();
+		FillSolidBlockTbls();
+		IncProgress();
+
+		if(lvldir == 5)
+			GetPortalLvlPos();
 
-		for (i = 0; i < MAXDUNX; i++) {
-			for (j = 0; j < MAXDUNY; j++)
-				dFlags[i][j] |= DFLAG_LIT;
+		for(i = 0; i < MAX_PLRS; i++) {
+			if(plr[i].plractive && currlevel == plr[i].plrlevel) {
+				InitPlayerGFX(i);
+				if(lvldir != 4)
+					InitPlayer(i, firstflag);
+			}
 		}
 
-		InitTowners();
-		InitItems();
-		InitMissiles();
+		InitMultiView();
 		IncProgress();
-		if (!firstflag && lvldir != 4 && plr[myplr]._pLvlVisited[currlevel]) {
-			if (gbMaxPlayers != 1)
-				goto LABEL_53;
+
+		if(firstflag || lvldir == 4 || !plr[myplr]._pSLvlVisited[setlvlnum]) {
+			InitItems();
+			SavePreLighting();
+		} else {
 			LoadLevel();
 		}
-		if (gbMaxPlayers == 1)
-			goto LABEL_54;
-	LABEL_53:
-		DeltaLoadLevel();
-		goto LABEL_54;
-	}
-	pSpeedCels = DiabloAllocPtr(0x100000);
-	LoadSetMap();
-	IncProgress();
-	GetLevelMTypes();
-	InitMonsters();
-	InitMissileGFX();
-	InitDead();
-	FillSolidBlockTbls();
-	IncProgress();
-	if (lvldir == 5)
-		GetPortalLvlPos();
-
-	for (i = 0; i < MAX_PLRS; i++) {
-		if (plr[i].plractive) {
-			if (currlevel == plr[i].plrlevel) {
-				InitPlayerGFX(v2);
-				if (lvldir != 4)
-					InitPlayer(v2, firstflag);
-			}
-		}
-		++v2;
-	}
 
-	InitMultiView();
-	IncProgress();
-	if (firstflag || lvldir == 4 || !plr[myplr]._pSLvlVisited[setlvlnum]) {
-		InitItems();
-		SavePreLighting();
-	} else {
-		LoadLevel();
+		InitMissiles();
+		IncProgress();
 	}
-	InitMissiles();
-	IncProgress();
-LABEL_72:
+
 	SyncPortals();
 
-	for (i = 0; i < MAX_PLRS; i++) {
-		if (plr[i].plractive && plr[i].plrlevel == currlevel && (!plr[i]._pLvlChanging || i == myplr)) {
-			if (plr[i]._pHitPoints <= 0)
-				dFlags[plr[i].WorldX][plr[i].WorldY] |= DFLAG_DEAD_PLAYER;
-			else if (gbMaxPlayers == 1)
-				dPlayer[plr[i].WorldX][plr[i].WorldY] = i + 1;
-			else
-				SyncInitPlrPos(i);
+	for(i = 0; i < MAX_PLRS; i++) {
+		if(plr[i].plractive && plr[i].plrlevel == currlevel && (!plr[i]._pLvlChanging || i == myplr)) {
+			if(plr[i]._pHitPoints > 0) {
+				if(gbMaxPlayers == 1)
+					dPlayer[plr[i].WorldX][plr[i].WorldY] = i + 1;
+				else
+					SyncInitPlrPos(i);
+			} else {
+				dFlags[plr[i].WorldX][plr[i].WorldY] |= 4;
+			}
 		}
 	}
 
-	if (leveltype != DTYPE_TOWN)
+	if(leveltype != DTYPE_TOWN)
 		SetDungeonMicros();
+
 	InitLightMax();
 	IncProgress();
 	IncProgress();
-	if (firstflag) {
+
+	if(firstflag) {
 		InitControlPan();
 		IncProgress();
 	}
-	if (leveltype != DTYPE_TOWN) {
+	if(leveltype != DTYPE_TOWN) {

```

Comment:
I start from scratch by dumping the function with an updated IDB and then work from that whilst comparing other patches (a bit more hassle, but it gets done right). And then format by hand to make it look good. I've never used unnecessary spaces after if/else or parenthesis. If you look at the old DX2 samples, they always use `for(int x)` or `for( int x )`.

Reply:
If it's how it was done in the dx samples I think it would be fair to adjust the code style to also follow that. I'll look in to it one of these days.